src/Aspire.Cli/Commands/PublishCommand.cs

```diff
@@ -18,17 +19,20 @@ internal sealed class PublishCommand : BaseCommand
     private readonly IDotNetCliRunner _runner;
     private readonly IInteractionService _interactionService;
     private readonly IProjectLocator _projectLocator;
+    private readonly IAppHostBuilder _appHostBuilder;
 
-    public PublishCommand(IDotNetCliRunner runner, IInteractionService interactionService, IProjectLocator projectLocator)
+    public PublishCommand(IDotNetCliRunner runner, IInteractionService interactionService, IProjectLocator projectLocator, IAppHostBuilder appHostBuilder)
         : base("publish", "Generates deployment artifacts for an Aspire app host project.")
     {
         ArgumentNullException.ThrowIfNull(runner);
         ArgumentNullException.ThrowIfNull(interactionService);
         ArgumentNullException.ThrowIfNull(projectLocator);
+        ArgumentNullException.ThrowIfNull(appHostBuilder);
```

Comment:
Ya know, this argument null checking isn't really necessary 😄. We not shipping a public API.